orbit-2-core/README.md

```diff
@@ -0,0 +1,319 @@
+not post any event back to the UI.
+
+## Including the module
+
+Orbit 2 is a modular framework. You will need this module to get started!
+
+Additional functionality is provided through plugins included in separate
+modules. Install only what you need!
+
+```kotlin
+implementation("com.babylon.orbit2:orbit-core:<latest-version>")
+```
+
+## Orbit container
+
+A [Container](src/main/java/com/babylon/orbit2/Container.kt) is the heart of
+the Orbit MVI system. It retains the state, allows you to listen to side effect
+and state updates and allows you to modify the state through the `orbit`
+function which executes DSL operators of your desired business logic.
+
+### ContainerHost
+
+A [ContainerHost](src/main/java/com/babylon/orbit2/ContainerHost.kt) is not
+strictly required to work with an Orbit
+[Container](src/main/java/com/babylon/orbit2/Container.kt), but it simplifies
+and organises access to it. A Host typically defines Orbit flows (chains of DSL
+operators to be invoked on the container) as functions that can be called by
+e.g. the UI.
+
+In a typical implementation one would subclass Android's `ViewModel` along with
+implementing `ContainerHost` in order to create an Orbit-enabled Android
+`ViewModel`.
+
+## DSL plugin registry
+
+Modules like `orbit-coroutines` or `orbit-rxjava2` contain plugins that extend
+Orbit's DSL. In order for Orbit to recognise these plugins, they need to be
+installed at runtime, like so:
+
+``` kotlin
+class MyApplication : Application() {
+    override fun onCreate() {
+        OrbitDslPlugins.register(
+            CoroutineDslPlugin,
+            RxJava2DslPlugin
+        )
+    }
+}
+```
+
+The core syntax DSL plugin (`transform/sideEffect/reduce`) works out of the box
+and does not need to be explicitly installed.
+
+## Core DSL Operators
+
+The Core DSL contains the following operators:
+
+- transform
+- sideEffect
+- reduce
+
+Transformers, side effects and reducers are invoked via simple DSL operators.
+Operators are invoked via the `orbit` function in a `ContainerHost` (or, less
+commonly, a `Container` directly)
+
+``` kotlin
+data class ExampleState(val seen: List<String> = emptyList())
+
+sealed class ExampleSideEffect {
+   data class Toast(val text: String)
+}
+
+class ExampleViewModel : ContainerHost<ExampleState, ExampleSideEffect>, ViewModel() {
+    override val container = Container.create<ExampleState, ExampleSideEffect>(ExampleState())
+
+    fun example(number: Int) = orbit {
+       transform {
+          number.toString()
+       }
+         .sideEffect { post(ExampleSideEffect.Toast(event)) }
+         .reduce { currentState.copy(seen = currentState.seen + event) }
+    }
+}
+```
+
+Aside from the three DSL functions already mentioned, more operators are
+provided via plugins providing e.g. support for `RxJava2` or Kotlin
+`Coroutines`. In theory Orbit can work with any async and streaming framework -
+plugins can be created easily.
```

Comment:
we don't really mention how any of these core operators run.. i.e. blocking the current thread or not